src/mixin/FlashLoanSolverWrapper.sol

```diff
@@ -56,34 +53,13 @@ abstract contract FlashLoanSolverWrapper is IFlashLoanSolverWrapper {
         require(token.approve(target, amount), "Approval failed");
     }
 
-    function triggerFlashLoan(address lender, IERC20 token, uint256 amount, bytes memory settlement) internal virtual;
-
-    function flashLoanCallback(bytes memory settlement) internal {
-        require(settlement.length > 0, "No settlement pending");
-        require(keccak256(settlement) == settlementHash, "Settlement hash not matching");
-        settlementHash = NO_SETTLEMENT;
-        (bool result,) = address(settlementContract).call(settlement);
-        // Todo: pass through error message.
-        require(result, "Settlement reverted");
-    }
+    function triggerFlashLoan(address lender, IERC20 token, uint256 amount, bytes memory callbackData)
+        internal
+        virtual;
 
-    /// @dev Extracts the Solidity ABI selector for the specified interaction.
-    ///
-    /// @param callData Interaction data.
-    /// @return result The 4 byte function selector of the call encoded in
-    /// this interaction.
-    function selector(bytes calldata callData) private pure returns (bytes4 result) {
-        if (callData.length >= 4) {
-            // NOTE: Read the first word of the calldata. The value does not
-            // need to be shifted since `bytesN` values are left aligned, and
-            // the value does not need to be masked since masking occurs when
-            // the value is accessed and not stored:
-            // <https://docs.soliditylang.org/en/v0.8.28/abi-spec.html#formal-specification-of-the-encoding>
-            // <https://docs.soliditylang.org/en/v0.8.26/assembly.html#access-to-external-variables-functions-and-libraries>
-            // solhint-disable-next-line no-inline-assembly
-            assembly {
-                result := calldataload(callData.offset)
-            }
-        }
+    function flashLoanCallback(bytes memory callbackData) internal {
+        require(keccak256(callbackData) == callbackDataHash, "Callback data hash not matching");
+        callbackDataHash = NO_DATA;
```

Comment:
Is this still required now if we check it in the router itself? Maybe the abstract solver wrapper is becoming a bit overkill in this design.